library/Fission/CLI/Command/Watch.hs

```diff
@@ -102,32 +103,14 @@ watcher runner Watch.Options {..} = do
 
       IPFS.addDir ignoredFiles absPath >>= putErrOr \cid@(CID hash) -> do
         UTF8.putText $ "👀 Watching " <> Text.pack absPath <> " for changes...\n"
-
-        sendRequestM (updateApp url cid copyFiles) >>= putErrOr \_ -> do
+        req <- App.update url cid copyFiles
+        retryOnStatus [status502, status504] 100 req >>= putErrOr \_ -> do
```

Comment:
Wait, does this work? Shouldn't this be...

```haskell
let req = App.update url cid copyFiles
retryOnStatus [status502, status504] 100 req
```

...because otherwise, you're just passing the result of the first request back through repeatedly, no?

Reply:
I thought the same thing at first. But `App.update` returns type `m (ClientM NoContent)`. So this is unwrapping the top `m` but isn't touching the actual `ClientM` monad

`retryOnStatus` runs the actual `sendRequest` which is the function that does the monadic action

`sendRequestM` is actually a function that unwraps the top layer of the monad before passing it to `sendRequest`
```haskell
sendRequestM :: MonadWebClient m => m (ClientM a) -> m (Either ClientError a)
sendRequestM clientAction = sendRequest =<< clientAction
```

Reply:
hoping I have this straight in my head, so let me know if that sounds right!

Reply:
Ah, yes I see — it's because of the type of `authClient`. We need the `m` because we want ambient access to config variables (for the auth part without having to manually pass them around).

This may be clearer with naming:

```haskell
App.mkUpdate

-- or

App.updateM
```

or similar

Reply:
ah cool I like it :+1: